org.eclipse.xtext.junit4/src/org/eclipse/xtext/junit4/ide/AbstractHierarchyBuilderTest.java

```diff
@@ -0,0 +1,196 @@
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.xtend2.lib.StringConcatenation;
+import org.eclipse.xtext.EcoreUtil2;
+import org.eclipse.xtext.ide.editor.hierarchy.AbstractHierarchyBuilder;
+import org.eclipse.xtext.ide.editor.hierarchy.IHierarchyBuilder;
+import org.eclipse.xtext.ide.editor.hierarchy.IHierarchyNode;
+import org.eclipse.xtext.ide.editor.hierarchy.IHierarchyNodeReference;
+import org.eclipse.xtext.junit4.validation.ValidationTestHelper;
+import org.eclipse.xtext.resource.EObjectAtOffsetHelper;
+import org.eclipse.xtext.resource.IResourceDescriptionsProvider;
+import org.eclipse.xtext.resource.XtextResource;
+import org.eclipse.xtext.resource.XtextResourceSet;
+import org.eclipse.xtext.ui.editor.findrefs.SimpleLocalResourceAccess;
+import org.eclipse.xtext.util.LazyStringInputStream;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.Functions.Function1;
+import org.eclipse.xtext.xbase.lib.Pair;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure1;
+import org.junit.Assert;
+
+/**
+ * @author kosyakov - Initial contribution and API
+ * @since 2.10
+ */
+@SuppressWarnings("deprecation")
+public abstract class AbstractHierarchyBuilderTest {
+	protected static class HierarchyBuilderTestConfiguration {
+		private Function1<? super ResourceSet, ? extends IHierarchyBuilder> hierarchyBuilderProvider;
+
+		private Collection<Pair<String, String>> models = new ArrayList<>();
+
+		private int index;
+
+		private String resourceURI;
+
+		private String expectedHierarchy;
+
+		public Function1<? super ResourceSet, ? extends IHierarchyBuilder> getHierarchyBuilderProvider() {
+			return hierarchyBuilderProvider;
+		}
+
+		public void setHierarchyBuilderProvider(Function1<? super ResourceSet, ? extends IHierarchyBuilder> hierarchyBuilderProvider) {
+			this.hierarchyBuilderProvider = hierarchyBuilderProvider;
+		}
+
+		public Collection<Pair<String, String>> getModels() {
+			return models;
+		}
+
+		public void setModels(Collection<Pair<String, String>> models) {
+			this.models = models;
+		}
+
+		public int getIndex() {
+			return index;
+		}
+
+		public void setIndex(int index) {
+			this.index = index;
+		}
+
+		public String getResourceURI() {
+			return resourceURI;
+		}
+
+		public void setResourceURI(String resourceURI) {
+			this.resourceURI = resourceURI;
+		}
+
+		public String getExpectedHierarchy() {
+			return expectedHierarchy;
+		}
+
+		public void setExpectedHierarchy(String expectedHierarchy) {
+			this.expectedHierarchy = expectedHierarchy;
+		}
+	}
+
+	@Inject
+	private ValidationTestHelper validationTestHelper;
+
+	@Inject
+	private EObjectAtOffsetHelper eObjectAtOffsetHelper;
+
+	@Inject
+	private Provider<XtextResourceSet> resourceSetProvider;
+
+	@Inject
+	private IResourceDescriptionsProvider resourceDescriptionsProvider;
+
+	protected void testBuildHierarchy(Procedure1<? super AbstractHierarchyBuilderTest.HierarchyBuilderTestConfiguration> configurator) {
```

Comment:
AbstractHierarchyBuilderTest.HierarchyBuilderTestConfiguration => HierarchyBuilderTestConfiguration
(remove superflues prefix at all places in this class).